src/classes/IQueryFilter.cls

```diff
@@ -14,7 +14,7 @@ public interface IQueryFilter {
 
     // Setter methods
     IQueryFilter filterByField(Schema.SObjectField fieldToFilter, QueryOperator operator, Object providedValue);
-    IQueryFilter filterByParentField(List<Schema.SObjectField> fieldToFilterChain, QueryOperator operator, Object providedValue);
+    IQueryFilter filterByParentField(SObjectFieldWrapper fieldWrapper, QueryOperator operator, Object providedValue);
```

Comment:
I'm trying to decide if it would be better if there is only a 'filterByField' method and eliminate filterByParentField. Seem reasonable? Or should I keep the current  version of `filterByField(Schema.SObjectField fieldToFilter, QueryOperator operator, Object providedValue);`? 

My thought is that when filtering on a single field, it's easier to do this...
`filterByField(Schema.Contact.Email, QueryOperator.NOT_EQUAL_TO, null);`
instead of this
`filterByField(new SObjectFieldWrapper(Schema.Contact.Email), QueryOperator.NOT_EQUAL_TO, null);`
On the other hand, I like the idea of there being just one filterByField method, like this `IQueryFilter filterByField(SObjectFieldWrapper fieldWrapper, QueryOperator operator, Object providedValue);`